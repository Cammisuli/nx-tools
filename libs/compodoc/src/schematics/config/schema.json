{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "compodoc-configure",
  "title": "Compodoc Configuration",
  "description": "Configure compodoc target for a project.",
  "type": "object",
  "properties": {
    "project": {
      "description": "Project name to add the target to.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Which project the target should be added to?"
    }
  },
  "required": ["project"]
}
