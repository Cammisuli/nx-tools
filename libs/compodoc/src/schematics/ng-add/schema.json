{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "compodoc-add",
  "title": "Add Compodoc",
  "description": "Install compodoc plugin."
}
