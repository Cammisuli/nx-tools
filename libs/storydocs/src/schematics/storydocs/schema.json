{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "Storydocs",
  "title": "",
  "type": "object",
  "properties": {
    "project": {
      "description": "Project name to add the architect to.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Which project the architect should be added to?"
    },
    "configureCypress": {
      "description": "Run the cypress-configure schematic",
      "type": "boolean",
      "x-prompt": "Configure a cypress e2e app to run against the storybook instance?"
    },
    "externalCompodoc": {
      "description": "Use an existing target for Compodoc.",
      "type": "boolean",
      "default": false
    },
    "externalStorybook": {
      "description": "Use an existing target for Storybook.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["project"]
}
