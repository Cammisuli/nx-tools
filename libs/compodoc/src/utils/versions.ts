export const versions = {
  '@twittwer/compodoc': '*',
  '@compodoc/compodoc': '*',
};
