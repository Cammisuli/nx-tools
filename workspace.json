{
  "version": 1,
  "projects": {
    "compodoc": {
      "root": "libs/compodoc",
      "sourceRoot": "libs/compodoc/src",
      "projectType": "library",
      "schematics": {},
      "architect": {
        "lint": {
          "builder": "@nrwl/linter:lint",
          "options": {
            "linter": "eslint",
            "config": "libs/compodoc/.eslintrc",
            "tsConfig": [
              "libs/compodoc/tsconfig.lib.json",
              "libs/compodoc/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/compodoc/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/compodoc/jest.config.js",
            "tsConfig": "libs/compodoc/tsconfig.spec.json",
            "passWithNoTests": true
          }
        },
        "build": {
          "builder": "@nrwl/node:package",
          "options": {
            "outputPath": "dist/libs/compodoc",
            "tsConfig": "libs/compodoc/tsconfig.lib.json",
            "packageJson": "libs/compodoc/package.json",
            "main": "libs/compodoc/src/index.ts",
            "assets": [
              "libs/compodoc/*.md",
              {
                "input": "./libs/compodoc/src",
                "glob": "**/*.!(ts)",
                "output": "./src"
              },
              {
                "input": "./libs/compodoc",
                "glob": "collection.json",
                "output": "."
              },
              {
                "input": "./libs/compodoc",
                "glob": "builders.json",
                "output": "."
              }
            ]
          }
        },
        "release": {
          "builder": "@nrwl/workspace:run-commands",
          "options": {
            "commands": [{ "command": "npx semantic-release" }],
            "cwd": "libs/compodoc"
          }
        }
      }
    },
    "compodoc-e2e": {
      "projectType": "application",
      "root": "apps/compodoc-e2e",
      "sourceRoot": "apps/compodoc-e2e/src",
      "architect": {
        "e2e": {
          "builder": "@nrwl/nx-plugin:e2e",
          "options": {
            "target": "compodoc:build",
            "npmPackageName": "@twittwer/compodoc",
            "pluginOutputPath": "dist/libs/compodoc",
            "jestConfig": "apps/compodoc-e2e/jest.config.js",
            "tsSpecConfig": "apps/compodoc-e2e/tsconfig.spec.json"
          }
        }
      }
    }
  },
  "cli": {
    "defaultCollection": "@nrwl/workspace"
  },
  "schematics": {
    "@nrwl/workspace": {
      "library": {
        "linter": "eslint"
      }
    },
    "@nrwl/cypress": {
      "cypress-project": {
        "linter": "eslint"
      }
    },
    "@nrwl/react": {
      "application": {
        "linter": "eslint"
      },
      "library": {
        "linter": "eslint"
      }
    },
    "@nrwl/next": {
      "application": {
        "linter": "eslint"
      }
    },
    "@nrwl/web": {
      "application": {
        "linter": "eslint"
      }
    },
    "@nrwl/node": {
      "application": {
        "linter": "eslint"
      },
      "library": {
        "linter": "eslint"
      }
    },
    "@nrwl/nx-plugin": {
      "plugin": {
        "linter": "eslint"
      }
    },
    "@nrwl/nest": {
      "application": {
        "linter": "eslint"
      }
    },
    "@nrwl/express": {
      "application": {
        "linter": "eslint"
      },
      "library": {
        "linter": "eslint"
      }
    }
  }
}
