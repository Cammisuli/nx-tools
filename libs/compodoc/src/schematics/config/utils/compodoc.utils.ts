import { Tree } from '@angular-devkit/schematics';
import { getProjectConfig } from './workspace.utils';
import { join } from 'path';
import { ProjectType } from '@nrwl/workspace';

export function getTsConfigForProject(tree: Tree, projectName: string): string {
  const { root: projectRoot, projectType } = getProjectConfig(
    tree,
    projectName,
  );

  let tsConfig = 'tsconfig.compodoc.json';
  if (!tree.exists(join(projectRoot, tsConfig))) {
    tsConfig =
      projectType === ProjectType.Application
        ? 'tsconfig.app.json'
        : 'tsconfig.lib.json';
  }
  if (!tree.exists(join(projectRoot, tsConfig))) {
    tsConfig = 'tsconfig.json';
  }
  if (!tree.exists(join(projectRoot, tsConfig))) {
    throw new Error(
      `Missing TSConfig: Cannot find a "tsconfig[.(compodoc|lib|app)].json" file in "${projectRoot}".`,
    );
  }
  return tsConfig;
}
