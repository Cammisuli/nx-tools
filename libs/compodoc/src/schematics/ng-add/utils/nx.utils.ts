import { Rule } from '@angular-devkit/schematics';
import { updateJsonInTree } from '@nrwl/workspace';

export function addCacheableOperation(operation: string): Rule {
  return updateJsonInTree('nx.json', nxJson => {
    if (
      !nxJson.tasksRunnerOptions?.default?.runner ||
      !Array.isArray(
        nxJson.tasksRunnerOptions?.default?.options?.cacheableOperations,
      )
    ) {
      return nxJson;
    }

    const isAlreadyCached = nxJson.tasksRunnerOptions.default.options.cacheableOperations.includes(
      operation,
    );
    if (!isAlreadyCached) {
      nxJson.tasksRunnerOptions.default.options.cacheableOperations.push(
        operation,
      );
    }

    return nxJson;
  });
}
