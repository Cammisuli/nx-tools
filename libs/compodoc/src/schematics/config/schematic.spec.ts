import { Tree } from '@angular-devkit/schematics';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { join } from 'path';
import { createEmptyWorkspace } from '@nrwl/workspace/testing';
import { CompodocConfigSchema } from './schema';
import { getProjectConfig } from './utils/workspace.utils';

describe('Compodoc "config" Schematic', () => {
  let testTree: Tree;

  const testRunner = new SchematicTestRunner(
    '@twittwer/compodoc',
    join(__dirname, '../../../collection.json'),
  );

  beforeEach(() => {
    testTree = Tree.empty();
    createEmptyWorkspace(testTree);
  });

  describe('for library', () => {
    beforeEach(async () => {
      testTree = await testRunner
        .runExternalSchematicAsync(
          // TODO: should use `@nrwl/angular`
          '@nrwl/angular',
          'library',
          { name: 'test-lib' },
          testTree,
        )
        .toPromise();
    });

    it('should add "compodoc" target to project', async () => {
      const options: Partial<CompodocConfigSchema> = {
        project: 'test-lib',
      };
      await testRunner
        .runSchematicAsync('config', options, testTree)
        .toPromise();

      const projectConfig = getProjectConfig(testTree, 'test-lib');
      expect(projectConfig.architect['compodoc']).toEqual({
        builder: '@twittwer/compodoc:compodoc',
        options: {
          outputPath: 'dist/compodoc/test-lib',
          tsConfig: 'libs/test-lib/tsconfig.lib.json',
        },
        configurations: {
          json: {
            exportFormat: 'json',
          },
        },
      });
    });

    describe('should prefer more specific tsconfig files', () => {
      const possibleTsConfigFiles = [
        'libs/test-lib/tsconfig.compodoc.json',
        'libs/test-lib/tsconfig.lib.json',
        'libs/test-lib/tsconfig.json',
      ];
      const tsConfigCombinations = [
        possibleTsConfigFiles.slice(0),
        possibleTsConfigFiles.slice(1),
        possibleTsConfigFiles.slice(2),
      ];
      it.each(tsConfigCombinations)('%s', async (...tsConfigFiles) => {
        possibleTsConfigFiles.forEach(file => {
          const isPartOfTestCase = tsConfigFiles.includes(file);
          const existsInSetup = testTree.exists(file);
          if (isPartOfTestCase && !existsInSetup) {
            testTree.create(file, '');
          }
          if (!isPartOfTestCase && existsInSetup) {
            testTree.delete(file);
          }
        });

        const expectedTsConfig = tsConfigFiles[0];
        await testRunner
          .runSchematicAsync('config', { project: 'test-lib' }, testTree)
          .toPromise();

        const projectConfig = getProjectConfig(testTree, 'test-lib');
        expect(projectConfig.architect['compodoc'].options.tsConfig).toBe(
          expectedTsConfig,
        );
      });
    });
  });

  it.todo('for application');
  it.todo('--workspaceDocs');
});
