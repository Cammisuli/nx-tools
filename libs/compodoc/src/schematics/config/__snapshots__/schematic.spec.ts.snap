// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compodoc "config" Schematic for application in nested directory --workspaceDocs tsconfig.compodoc.json should create a compodoc specific tsconfig file 1`] = `
Object {
  "extends": "./tsconfig.app.json",
  "include": Array [
    "../../../apps/**/*.ts",
    "../../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for application in nested directory --workspaceDocs tsconfig.compodoc.json should extend most specific tsconfig files tsconfig.app.json 1`] = `
Object {
  "extends": "./tsconfig.app.json",
  "include": Array [
    "../../../apps/**/*.ts",
    "../../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for application in nested directory --workspaceDocs tsconfig.compodoc.json should extend most specific tsconfig files tsconfig.json 1`] = `
Object {
  "extends": "./tsconfig.json",
  "include": Array [
    "../../../apps/**/*.ts",
    "../../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for application in undefined directory --workspaceDocs tsconfig.compodoc.json should create a compodoc specific tsconfig file 1`] = `
Object {
  "extends": "./tsconfig.app.json",
  "include": Array [
    "../../apps/**/*.ts",
    "../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for application in undefined directory --workspaceDocs tsconfig.compodoc.json should extend most specific tsconfig files tsconfig.app.json 1`] = `
Object {
  "extends": "./tsconfig.app.json",
  "include": Array [
    "../../apps/**/*.ts",
    "../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for application in undefined directory --workspaceDocs tsconfig.compodoc.json should extend most specific tsconfig files tsconfig.json 1`] = `
Object {
  "extends": "./tsconfig.json",
  "include": Array [
    "../../apps/**/*.ts",
    "../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for library in nested directory --workspaceDocs tsconfig.compodoc.json should create a compodoc specific tsconfig file 1`] = `
Object {
  "extends": "./tsconfig.lib.json",
  "include": Array [
    "../../../apps/**/*.ts",
    "../../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for library in nested directory --workspaceDocs tsconfig.compodoc.json should extend most specific tsconfig files tsconfig.json 1`] = `
Object {
  "extends": "./tsconfig.json",
  "include": Array [
    "../../../apps/**/*.ts",
    "../../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for library in nested directory --workspaceDocs tsconfig.compodoc.json should extend most specific tsconfig files tsconfig.lib.json 1`] = `
Object {
  "extends": "./tsconfig.lib.json",
  "include": Array [
    "../../../apps/**/*.ts",
    "../../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for library in undefined directory --workspaceDocs tsconfig.compodoc.json should create a compodoc specific tsconfig file 1`] = `
Object {
  "extends": "./tsconfig.lib.json",
  "include": Array [
    "../../apps/**/*.ts",
    "../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for library in undefined directory --workspaceDocs tsconfig.compodoc.json should extend most specific tsconfig files tsconfig.json 1`] = `
Object {
  "extends": "./tsconfig.json",
  "include": Array [
    "../../apps/**/*.ts",
    "../../libs/**/*.ts",
  ],
}
`;

exports[`Compodoc "config" Schematic for library in undefined directory --workspaceDocs tsconfig.compodoc.json should extend most specific tsconfig files tsconfig.lib.json 1`] = `
Object {
  "extends": "./tsconfig.lib.json",
  "include": Array [
    "../../apps/**/*.ts",
    "../../libs/**/*.ts",
  ],
}
`;
